### ISSUE TITLE
If a cart expires and you start a new cart you cannot checkout, reports the cart has expired.

### ISSUE THREAD

**Issue by philipr-za** [big-neon/bn-api#356]

**Comment by Timo614** [comment]
It looks like we still allow new tickets to be added to the cart as well in that state.

I guess it still makes sense for us to allow the cart to be seen after expiration as you'd want to be able to report it was expired to the user until they go ahead and start a new cart.

So thinking:
1. Allow the cart endpoint to still return an expired cart so that we can show it's expired to customers on the frontend
2. The add cart logic should create a brand new cart in the event the existing cart is expired
3. If the cart is expired the frontend shouldn't allow the customer to checkout
4. It may be worth having an endpoint to create a new cart by itself so the frontend can hit that to clear the cart (ie: show the expired cart but then allow the customer to press some button to hide the expired cart / refresh with a new cart)

@Jasonvdb what were you thinking for behavior here on the frontend? Does the above make sense? Or would it be best to just never return an expired cart and always return a new cart post expiration? Not sure if it would be useful to be able to see the cart that had expired prior to them taking a new action / starting a new cart or not.

For now I'll do the second item in the list as it seems like we'd never want to allow an expired cart to have new items added to it. If we want to display the expired cart still I'll also do the fourth item if not I'll just have the cart endpoint always return a new cart on expiration.

**Comment by Timo614** [comment]
Actually I guess some of this depends on how we want to treat these draft orders. If they expire should we just clear them and set a new expiration? @mikethetike is that what we were thinking? Or just keep the draft order around but don't surface it to the customer on expiration?

**Comment by Jasonvdb** [comment]
If the `/cart` endpoint is returning the time till cart expiry I can watch that in bn-web and do a simple dialog to the user for when it's expired. I don't think they need much more than that.

So no need to return an expired cart. Adding to an expired cart should just create a new cart and scrap the old one. Once a cart is expired then getting cart details from `/cart` can just be empty.

When a user starts the process I can display a countdown timer, most ticketing platforms seem to do that. So if a user finds their cart expired it's not like they weren't warned.

In summary I think just your point 2 is necessary then? @mikethetike what do you think?

**Comment by mikethetike** [comment]
I agree, when the cart expires, scrap it and create a new one. This will allow us to draw a report in future that tells us how many of these carts expired (or were abandoned). (So agree that only point 2 is relevant)

@zmelamed Any comments?

**Comment by Timo614** [comment]
Thinking it over right now I think if you add a cart item, remove that item, and then check the cart it'll still have a date toward expiration. I guess in that case we should also remove the cart entirely from the db as it'll have no relevant information for a report and probably just confuse the user purchasing to see their empty cart with a timer on it.
